### PR TITLE
Claims Challenge user story

### PR DESCRIPTION
**Pull request by SomkaPe**
Claims challenge user story https://github.com/AzureAD/azure-activedirectory-library-for-java/issues/159

### CODE REVIEW

File: src/main/java/com/microsoft/aad/adal4j/AdalTokenRequest.java [hunk 1]
```diff
@@ -104,13 +106,28 @@ AuthenticationResult executeOAuthRequestAndProcessResponse()
...
+            }
         }
 
         return result;
     }
 
+    private String getClaims(String httpResponseContentStr) {
+        JsonElement root = new JsonParser().parse(httpResponseContentStr);
+
+        String claims = root.getAsJsonObject().get("claims").getAsString();
```
Comment (kpanwar) [inline src/main/java/com/microsoft/aad/adal4j/AdalTokenRequest.java]: .getAsString() can throw null pointer.

File: src/samples/web-app-samples-for-adal4j/src/main/webapp/WEB-INF/web.xml [hunk 2]
```diff
@@ -4,27 +4,56 @@
...
+		<param-value>XXX</param-value>
+	</context-param>
+	<context-param>
+		<param-name>obo_application_id_uri</param-name>
+		<param-value>XXX</param-value>
 	</context-param>
 
+	<context-param>
+		<param-name>mfa_protected_api_id_uri</param-name>
+		<param-value>XXX</param-value>
```
Comment (kpanwar) [inline src/samples/web-app-samples-for-adal4j/src/main/webapp/WEB-INF/web.xml]: internal scan will fail "XXX" as a dirty word

Comment (SomkaPe) [inline src/samples/web-app-samples-for-adal4j/src/main/webapp/WEB-INF/web.xml]: done

File: src/main/java/com/microsoft/aad/adal4j/AdalTokenRequest.java (hunk 1 above)
Comment (SomkaPe) [inline src/main/java/com/microsoft/aad/adal4j/AdalTokenRequest.java]: changed

File: src/main/java/com/microsoft/aad/adal4j/AdalClaimsChallengeException.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A
+ * PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ * See the Apache License, Version 2.0 for the specific language
+ * governing permissions and limitations under the License.
+ ******************************************************************************/
+
+package com.microsoft.aad.adal4j;
+
+public class AdalClaimsChallengeException extends AuthenticationException {
```
Comment (kpanwar) [inline src/main/java/com/microsoft/aad/adal4j/AdalClaimsChallengeException.java]: missing public header documentation

File: src/main/java/com/microsoft/aad/adal4j/AdalClaimsChallengeException.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    public String getClaims() {
```
Comment (kpanwar) [inline src/main/java/com/microsoft/aad/adal4j/AdalClaimsChallengeException.java]: missing method header doc

File: src/main/java/com/microsoft/aad/adal4j/AdalClaimsChallengeException.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        return claims;
+    }
+
+    private final String claims;
+
+
+    public AdalClaimsChallengeException(String message, String claims) {
```
Comment (kpanwar) [inline src/main/java/com/microsoft/aad/adal4j/AdalClaimsChallengeException.java]: missing method header doc

File: src/main/java/com/microsoft/aad/adal4j/AdalClaimsChallengeException.java (hunk 3 above)
Comment (SomkaPe) [inline src/main/java/com/microsoft/aad/adal4j/AdalClaimsChallengeException.java]: added

File: src/main/java/com/microsoft/aad/adal4j/AdalClaimsChallengeException.java (hunk 4 above)
Comment (SomkaPe) [inline src/main/java/com/microsoft/aad/adal4j/AdalClaimsChallengeException.java]: added

File: src/main/java/com/microsoft/aad/adal4j/AdalClaimsChallengeException.java (hunk 5 above)
Comment (SomkaPe) [inline src/main/java/com/microsoft/aad/adal4j/AdalClaimsChallengeException.java]: added

**Pull request merged**